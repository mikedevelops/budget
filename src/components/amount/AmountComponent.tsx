import * as React from 'react';
import { IAmountComponentPropsInterface } from './AmountComponentInterface';

const AmountComponent: React.SFC<IAmountComponentPropsInterface> = ({
    currency,
    value
}) => {
    return <p className="amount">{ currency }{ value }</p>;
};

export default AmountComponent;
