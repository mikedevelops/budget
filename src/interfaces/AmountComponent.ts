export interface AmountComponentPropsInterface {
    currency: string;
    value: number;
}
