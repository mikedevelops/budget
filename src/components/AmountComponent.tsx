import * as React from 'react';
import { AmountComponentPropsInterface } from '../interfaces/AmountComponent';

const Target: React.SFC<AmountComponentPropsInterface> = ({
    currency,
    value
}) => {
    return <p>{ currency }{ value }</p>
}

export default Target;
