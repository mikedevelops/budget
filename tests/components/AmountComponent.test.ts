import * as React from 'react';
import * as renderer from 'react-test-renderer';
import AmountComponent from '../../src/components/AmountComponent';

describe('AmountComponent', () => {
    test('Should render an amount component', () => {
        const component = renderer.create(
            React.createElement(AmountComponent, { currency: '$', value: 666 })
        ).toJSON();

        expect(component).toMatchSnapshot();
    });
});
