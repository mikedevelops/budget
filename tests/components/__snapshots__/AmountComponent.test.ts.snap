// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountComponent Should render an amount component 1`] = `
<p>
  $
  666
</p>
`;
